.github/workflows/reusable-build.yml

```diff
@@ -73,6 +73,49 @@ jobs:
               $dnf install -y dnf-plugins-core
             fi
 
+            coreos_kernel () {
+              coreos_version=${1}
+              image_linux=$(skopeo inspect docker://quay.io/fedora/fedora-coreos:${coreos_version} | jq -r '.Labels["ostree.linux"]')
+              major_minor_patch=$(echo $image_linux | grep -oP '^\d+\.\d+\.\d+')
+              kernel_rel_part=$(echo $image_linux | grep -oP '^\d+\.\d+\.\d+\-\K([123][0]{2})')
+              arch=$(echo $image_linux | grep -oP 'fc\d+\.\K.*$')
+
+              kernel_rel="$kernel_rel_part.fc${{ matrix.fedora_version }}"
+              kernel_version="$major_minor_patch-$kernel_rel.$arch"
+              URL="https://kojipkgs.fedoraproject.org/packages/kernel/"$major_minor_patch"/"$kernel_rel"/"$arch"/kernel-"$kernel_version".rpm"
+              echo "Querying koji for ${coreos_version} kernel: $kernel_version"
+              echo "$URL"
+              HTTP_RESP=$(curl -sI "$URL" | grep ^HTTP)
+              if grep -qv "200 OK" <<< "${HTTP_RESP}"; then
+                echo "Koji failed to find $coreos_version kernel: $kernel_version"
+                case "$kernel_rel_part" in
+                  "300")
+                    kernel_rel_part="200"
```

Comment:
This will make it so uCore will kernel swap. This maybe an issue since we're then swapping off of coreOS kernel.

Reply:
We chatted a bit about this out of band, but to bring the conversation here... I don't think this causes a swap.

The first thing that happens is we construct a Koji URL from the `$image_linux` kernel version and verify the packages for that kernel exist in koji.

This should ALWAYS succeed unless:
1) koji is down/having problems
2) the fedora-coreos image is very messed up
3) we are trying to create a kernel cache for same kernel version but a previous fedora release (the use case which prompted this effort)

When we fail to find a matching kernel, it should really indicate that we've decremented the fedora release to and the version string doesn't match what was in the coreos image.

In that case, we decrement the "kernel release" number... either 300 > 200 or 200 > 100, and then we try Koji again.

If we fail a second time, it all fails.

I suppose we could try a third if we wanted to support 2 releases back, but that's more support than Fedora provides, so it very much seems like overkill.

Reply:
Yeah, I see where my logic was flawed. I had assumed that kernel version was still something different. We're good!